### ISSUE TITLE
[Setup Error] ImportError: libcublas.so.10.0: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by Frank-Dz** [pmh47/dirt#61]
After running 
`sudo pip3 install .`
I got:

Processing /home/frank/Documents/dirt
Requirement already satisfied (use --upgrade to upgrade): tensorflow-gpu>=1.6 in /usr/local/lib/python3.5/dist-packages (from dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): numpy in /home/frank/.local/lib/python3.5/site-packages (from dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): absl-py>=0.1.6 in /usr/local/lib/python3.5/dist-packages (from tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): keras-applications>=1.0.6 in /usr/local/lib/python3.5/dist-packages (from tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): six>=1.10.0 in /home/frank/.local/lib/python3.5/site-packages (from tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): protobuf>=3.6.1 in /usr/local/lib/python3.5/dist-packages (from tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): astor>=0.6.0 in /usr/local/lib/python3.5/dist-packages (from tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): tensorboard<1.14.0,>=1.13.0 in /usr/local/lib/python3.5/dist-packages (from tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): gast>=0.2.0 in /usr/local/lib/python3.5/dist-packages (from tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): termcolor>=1.1.0 in /usr/local/lib/python3.5/dist-packages (from tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): wheel>=0.26 in /usr/lib/python3/dist-packages (from tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): tensorflow-estimator<1.14.0rc0,>=1.13.0 in /usr/local/lib/python3.5/dist-packages (from tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): keras-preprocessing>=1.0.5 in /usr/local/lib/python3.5/dist-packages (from tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): grpcio>=1.8.6 in /usr/local/lib/python3.5/dist-packages (from tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): h5py in /usr/local/lib/python3.5/dist-packages (from keras-applications>=1.0.6->tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): setuptools in /home/frank/.local/lib/python3.5/site-packages (from protobuf>=3.6.1->tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): werkzeug>=0.11.15 in /usr/local/lib/python3.5/dist-packages (from tensorboard<1.14.0,>=1.13.0->tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): markdown>=2.6.8 in /usr/local/lib/python3.5/dist-packages (from tensorboard<1.14.0,>=1.13.0->tensorflow-gpu>=1.6->dirt===0.3.0)
Requirement already satisfied (use --upgrade to upgrade): mock>=2.0.0 in /usr/local/lib/python3.5/dist-packages (from tensorflow-estimator<1.14.0rc0,>=1.13.0->tensorflow-gpu>=1.6->dirt===0.3.0)
Installing collected packages: dirt
  Running setup.py install for dirt ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-vh7g5khw-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-75tdcapy-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    -- The CXX compiler identification is GNU 5.4.0
    -- The CUDA compiler identification is NVIDIA 10.0.130
    -- Check for working CXX compiler: /usr/bin/c++
    -- Check for working CXX compiler: /usr/bin/c++ - works
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Check for working CUDA compiler: /usr/local/cuda-10.0/bin/nvcc
    -- Check for working CUDA compiler: /usr/local/cuda-10.0/bin/nvcc - works
    -- Detecting CUDA compiler ABI info
    -- Detecting CUDA compiler ABI info - done
    -- Detecting CUDA compile features
    -- Detecting CUDA compile features - done
    -- Found OpenGL: /usr/lib/nvidia-418/libOpenGL.so  found components: OpenGL EGL
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/usr/local/lib/python2.7/dist-packages/tensorflow/__init__.py", line 24, in <module>
        from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import
      File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/__init__.py", line 49, in <module>
        from tensorflow.python import pywrap_tensorflow
      File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow.py", line 74, in <module>
        raise ImportError(msg)
    ImportError: Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow.py", line 58, in <module>
        from tensorflow.python.pywrap_tensorflow_internal import *
      File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow_internal.py", line 28, in <module>
        _pywrap_tensorflow_internal = swig_import_helper()
      File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow_internal.py", line 24, in swig_import_helper
        _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
    ImportError: libcublas.so.10.0: cannot open shared object file: No such file or directory
    
    
    Failed to load the native TensorFlow runtime.
    
    See https://www.tensorflow.org/install/errors
    
    for some common reasons and solutions.  Include the entire stack trace
    above this error message when asking for help.
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/usr/local/lib/python2.7/dist-packages/tensorflow/__init__.py", line 24, in <module>
        from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import
      File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/__init__.py", line 49, in <module>
        from tensorflow.python import pywrap_tensorflow
      File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow.py", line 74, in <module>
        raise ImportError(msg)
    ImportError: Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow.py", line 58, in <module>
        from tensorflow.python.pywrap_tensorflow_internal import *
      File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow_internal.py", line 28, in <module>
        _pywrap_tensorflow_internal = swig_import_helper()
      File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow_internal.py", line 24, in swig_import_helper
        _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
    ImportError: libcublas.so.10.0: cannot open shared object file: No such file or directory
    
    
    Failed to load the native TensorFlow runtime.
    
    See https://www.tensorflow.org/install/errors
    
    for some common reasons and solutions.  Include the entire stack trace
    above this error message when asking for help.
    CMake Error at CMakeLists.txt:27 (message):
      cannot find either cuda_launch_config.h or gpu_launch_config.h
    
    
    -- Configuring incomplete, errors occurred!
    See also "/tmp/pip-vh7g5khw-build/build/CMakeFiles/CMakeOutput.log".
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-vh7g5khw-build/setup.py", line 50, in <module>
        'Programming Language :: Python :: 3.7',
      File "/home/frank/.local/lib/python3.5/site-packages/setuptools/__init__.py", line 144, in setup
        return distutils.core.setup(**attrs)
      File "/usr/lib/python3.5/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/usr/lib/python3.5/distutils/dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/home/frank/.local/lib/python3.5/site-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/usr/lib/python3.5/distutils/command/install.py", line 583, in run
        self.run_command('build')
      File "/usr/lib/python3.5/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/tmp/pip-vh7g5khw-build/setup.py", line 24, in run
        build_csrc()
      File "/tmp/pip-vh7g5khw-build/setup.py", line 18, in build_csrc
        subprocess.check_call(['cmake', os.path.join(base_path, 'csrc')], cwd=build_path)
      File "/usr/lib/python3.5/subprocess.py", line 581, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['cmake', '/tmp/pip-vh7g5khw-build/csrc']' returned non-zero exit status 1
    
    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-vh7g5khw-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-75tdcapy-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-vh7g5khw-build/
You are using pip version 8.1.1, however version 20.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

**Comment by Frank-Dz** [comment]
- OS: Ubuntu16.04 LTS
- cmake: 3.17
- tensorflow-gpu==1.13.1
- cuda: 10.0
- cudnn: 7.6.4.38

**Comment by Frank-Dz** [comment]
Actually, the problem is that my python version is 3.5, yet the setup seems like to use py2.7.

**Comment by Frank-Dz** [comment]
Sorry, I found the error is:

```
 -- Detecting CUDA compiler ABI info
    -- Detecting CUDA compiler ABI info - done
    -- Detecting CUDA compile features
    -- Detecting CUDA compile features - done
    CMake Error at /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
      Could NOT find OpenGL (missing: OPENGL_opengl_LIBRARY)
    Call Stack (most recent call first):
      /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:445 (_FPHSA_FAILURE_MESSAGE)
      /usr/share/cmake-3.17/Modules/FindOpenGL.cmake:427 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
      CMakeLists.txt:6 (find_package)
    
    
    -- Configuring incomplete, errors occurred!
    See also "/tmp/pip-c11kbwig-build/build/CMakeFiles/CMakeOutput.log".
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-c11kbwig-build/setup.py", line 50, in <module>
        'Programming Language :: Python :: 3.7',
      File "/home/frank/.local/lib/python3.5/site-packages/setuptools/__init__.py", line 144, in setup
        return distutils.core.setup(**attrs)
      File "/usr/lib/python3.5/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/usr/lib/python3.5/distutils/dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/home/frank/.local/lib/python3.5/site-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/usr/lib/python3.5/distutils/command/install.py", line 583, in run
        self.run_command('build')
      File "/usr/lib/python3.5/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/tmp/pip-c11kbwig-build/setup.py", line 24, in run
        build_csrc()
      File "/tmp/pip-c11kbwig-build/setup.py", line 18, in build_csrc
        subprocess.check_call(['cmake', os.path.join(base_path, 'csrc')], cwd=build_path)
      File "/usr/lib/python3.5/subprocess.py", line 581, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['cmake', '/tmp/pip-c11kbwig-build/csrc']' returned non-zero exit status 1

```
I searched and installed all necessary pkgs and I also found that I do not have a fold like: 
'/usr/lib/nvidia-410' although I am using 410 driver.

**Comment by pmh47** [comment]
Please run `ls -l /usr/lib/*/*GL*` and paste the output here, to check if the nvidia opengl libraries exist. If you used the nvidia runfile installer for the driver, try reinstalling it -- it's possible that some other package has overwritten libEGL and others. The `/usr/lib/nvidia-XXX' folder exists only if you use the ubuntu-packaged version of the driver, so it is probably ok that it is not there.

**Comment by Frank-Dz** [comment]
> Please run `ls -l /usr/lib/*/*GL*` and paste the output here, to check if the nvidia opengl libraries exist. If you used the nvidia runfile installer for the driver, try reinstalling it -- it's possible that some other package has overwritten libEGL and others. The `/usr/lib/nvidia-XXX' folder exists only if you use the ubuntu-packaged version of the driver, so it is probably ok that it is not there.

OK! Thanks! But I met another problem:
```
2020-04-16 11:39:05.592220: F /home/frank/Documents/dirt/csrc/gl_common.h:46] extensions eglQueryDevicesEXT, eglQueryDeviceAttribEXT and eglGetPlatformDisplayEXT not available
```
I read a similar issue and tried their solutions, and it did not work for me.

[https://github.com/pmh47/dirt/issues/13](https://github.com/pmh47/dirt/issues/13)
> Thanks for your help! The LD_PRELOAD trick works well
> `LD_PRELOAD=/path/to/new/libEGL.so.1.1.0 tests/square_test.py`

Hi~
I also met the same problem.
I installed nvidia-driver, and got corresponding pkgs:
in **/usr/lib/nvidia-410**:
alt_ld.so.conf                 libnvidia-compiler.so
bin                            libnvidia-compiler.so.1
ld.so.conf                     libnvidia-compiler.so.410.78
libEGL_nvidia.so.0             libnvidia-eglcore.so.410.78
libEGL_nvidia.so.410.78        libnvidia-egl-wayland.so.1
libEGL.so                      libnvidia-egl-wayland.so.1.1.0
libEGL.so.1                    libnvidia-encode.so
libEGL.so.1.1.0                libnvidia-encode.so.1
libEGL.so.410.78               libnvidia-encode.so.410.78
libGLdispatch.so.0             libnvidia-fatbinaryloader.so.410.78
libGLESv1_CM_nvidia.so.1       libnvidia-fbc.so
libGLESv1_CM_nvidia.so.410.78  libnvidia-fbc.so.1
libGLESv1_CM.so                libnvidia-fbc.so.410.78
libGLESv1_CM.so.1              libnvidia-glcore.so.410.78
libGLESv1_CM.so.1.2.0          libnvidia-glsi.so.410.78
libGLESv2_nvidia.so.2          libnvidia-glvkspirv.so.410.78
libGLESv2_nvidia.so.410.78     libnvidia-ifr.so
libGLESv2.so                   libnvidia-ifr.so.1
libGLESv2.so.2                 libnvidia-ifr.so.410.78
libGLESv2.so.2.1.0             libnvidia-ml.so
libGL.so                       libnvidia-ml.so.1
libGL.so.1                     libnvidia-ml.so.410.78
libGL.so.1.7.0                 libnvidia-ptxjitcompiler.so.1
libGL.so.410.78                libnvidia-ptxjitcompiler.so.410.78
libGLX_indirect.so.0           libnvidia-rtcore.so.410.78
libGLX_nvidia.so.0             libnvidia-tls.so.410.78
libGLX_nvidia.so.410.78        libnvidia-wfb.so.1
libGLX.so                      libnvidia-wfb.so.410.78
libGLX.so.0                    libnvoptix.so.1
libnvcuvid.so                  libnvoptix.so.410.78
libnvcuvid.so.1                libOpenGL.so
libnvcuvid.so.410.78           libOpenGL.so.0
libnvidia-cbl.so.410.78        tls
libnvidia-cfg.so               vdpau
libnvidia-cfg.so.1             xorg
libnvidia-cfg.so.410.78


And I set 
```
export LD_PRELOAD=/usr/lib/nvidia-410/libEGL.so.1.1.0
```
in ~/.bashrc

but I still get this error:

2020-04-16 11:39:05.592220: F /home/frank/Documents/dirt/csrc/gl_common.h:46] extensions eglQueryDevicesEXT, eglQueryDeviceAttribEXT and eglGetPlatformDisplayEXT not available

Can anyone help?
Thanks in advance!
